### PR TITLE
Feature: Hide block signal tools by default

### PR DESCRIPTION
**Pull request by nielsmh**
Please stop building block signals and especially pre-signals. They're for the most part a bad idea and your networks will function much better with PBS. Let's hide it behind an Expert level setting.

### CODE REVIEW

File: src/rail_gui.cpp [hunk 1]
```diff
@@ -1552,6 +1557,25 @@ struct BuildSignalWindow : public PickerWindowBase {
...
+				case WID_BS_ELECTRIC_EXIT:
+				case WID_BS_ELECTRIC_COMBO:
+					*size = Dimension{ 0, 0 };
+					*resize = *size;
+					*fill = *size;
+					break;
+				default:
+					break;
+			}
+		}
```
Comment (PeterN) [inline src/rail_gui.cpp:1578]: Isn't there a better way to hide widgets? Planes or something?

Comment (nielsmh) [reply src/rail_gui.cpp:1578]: The only planes I've had success finding anything about in OTTD are airplanes, but there may be some corner of NUI that I still haven't discovered exists.

Comment (nielsmh) [reply src/rail_gui.cpp:1578]: Yeah an attempt at wrapping the buttons with an NWID_SELECTION on each row caused massive problems, it's much more trouble than worth to do it "the right way". May as well re-do the entire signals UI then.

File: src/lang/english.txt [hunk 2]
```diff
@@ -1556,6 +1556,8 @@ STR_CONFIG_SETTING_SEMAPHORE_BUILD_BEFORE_DATE                  :Automatically b
 STR_CONFIG_SETTING_SEMAPHORE_BUILD_BEFORE_DATE_HELPTEXT         :Set the year when electric signals will be used for tracks. Before this year, non-electric signals will be used (which have the exact same function, but different looks)
 STR_CONFIG_SETTING_ENABLE_SIGNAL_GUI                            :Enable the signal GUI: {STRING2}
 STR_CONFIG_SETTING_ENABLE_SIGNAL_GUI_HELPTEXT                   :Display a window for choosing signal types to build, instead of only window-less signal-type rotation with Ctrl+clicking on built signals
+STR_CONFIG_SETTING_SHOW_BLOCKSIGNALS                            :Enable building block signals: {STRING2}
+STR_CONFIG_SETTING_SHOW_BLOCKSIGNALS_HELPTEXT                   :Show the tools for building block signals and pre-signals. Usage of block signals and path signals is discouraged, since path signals usually solve the same problems better. Existing block signals will continue working.
```
Comment (LordAro) [inline src/lang/english.txt:1560]: Not a fan of this, reads as very disparaging to the user. Something like "Block & pre(not path!) signals are intended for advanced users only"

File: src/lang/english.txt [hunk 3]
```diff
@@ -1556,6 +1556,8 @@ STR_CONFIG_SETTING_SEMAPHORE_BUILD_BEFORE_DATE                  :Automatically b
 STR_CONFIG_SETTING_SEMAPHORE_BUILD_BEFORE_DATE_HELPTEXT         :Set the year when electric signals will be used for tracks. Before this year, non-electric signals will be used (which have the exact same function, but different looks)
 STR_CONFIG_SETTING_ENABLE_SIGNAL_GUI                            :Enable the signal GUI: {STRING2}
 STR_CONFIG_SETTING_ENABLE_SIGNAL_GUI_HELPTEXT                   :Display a window for choosing signal types to build, instead of only window-less signal-type rotation with Ctrl+clicking on built signals
+STR_CONFIG_SETTING_SHOW_BLOCKSIGNALS                            :Enable building block signals: {STRING2}
```
Comment (LordAro) [inline src/lang/english.txt:1559]: Maybe "Enable block signal GUI", as per "Enable the signal GUI" above. Or perhaps it should just be rolled into the same setting and be made a tri-state

**Review by LordAro — CHANGES_REQUESTED** [review]

**Comment by nielsmh** [comment]
Honestly this is a poor solution to the wrong problem, I'll rather remake the signals UI entirely.

**Pull request closed without merging**